### PR TITLE
feat: Add timeout and cleanUp to Amplitude client

### PR DESCRIPTION
**Pull request by bohan-amplitude**
### Summary

Add more logs when catching exception
trigger call backs when catching exception
Add setFlushTimeout method
Add shutdown interface to close client and release resources

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-Java/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  no

### CODE REVIEW

File: src/main/java/com/amplitude/Amplitude.java [hunk 1]
```diff
@@ -280,6 +293,15 @@ public void setRecordThrottledId(boolean record) {
     httpTransport.setRecordThrottledId(record);
   }
 
+  /**
+   * Release resource by:
+   * Terminate all threads for flushing events
+   * All events hold by these threads will trigger callbacks if amplitude client have configured callbacks method
+   */
+  public synchronized void cleanUp() throws InterruptedException {
```
Comment (justin-fiedler) [inline src/main/java/com/amplitude/Amplitude.java:301]: Should we name this "shudown" to match Ampli SDK?

File: src/main/java/com/amplitude/HttpTransport.java [hunk 2]
```diff
@@ -105,11 +145,10 @@ private CompletableFuture<Response> sendEvents(List<Event> events) {
           Response response = null;
           try {
             response = httpCall.makeRequest(events);
-            logger.debug("SEND", events, response);
+            logger.debug("SEND", "Events count " + events.size());
+            logger.debug("RESPONSE", response.toString());
           } catch (AmplitudeInvalidAPIKeyException e) {
             throw new CompletionException(e);
-          } catch (Exception e) {
```
Comment (liuyang1520) [inline src/main/java/com/amplitude/HttpTransport.java:111]: Just to double check, any reason that we want to remove this safe catch?

Comment (bohan-amplitude) [reply src/main/java/com/amplitude/HttpTransport.java:111]: moved to outer layer that call sendEvents() so I can handle unexpected exceptions in one place.

**Review by liuyang1520 — APPROVED** [review]
Except the minor doc style, other changes LGTM!

**Pull request merged**